MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.dev.cc

```diff
@@ -0,0 +1,141 @@
+      out[index*3 + 2] = magneticFieldAtPoint(in[index]);
+    }
+  }
+};
+
+int main() {
+  // get the list of devices on the current platform
+  auto const& devices = cms::alpakatools::devices<Platform>();
+  if (devices.empty()) {
+    std::cerr << "No devices available for the " EDM_STRINGIZE(ALPAKA_ACCELERATOR_NAMESPACE) " backend, "
+      "the test will be skipped.\n";
+    exit(EXIT_FAILURE);
+  }
+
+  ifstream file;
+  edm::FileInPath mydata("MagneticField/Engine/data/Regression/referenceField_160812_RII_3_8T.bin");
+  file.open(mydata.fullPath().c_str(), ios::binary);
+
+  int count = 0;
+  float px, py, pz;
+  float bx, by, bz;
+  vector<Vector3f> points;
+  vector<GlobalVector> referenceB_vec;
+
+  int numberOfPoints = 100;
+  do {
+    if (!(file.read((char*)&px, sizeof(float)) && file.read((char*)&py, sizeof(float)) &&
+          file.read((char*)&pz, sizeof(float)) && file.read((char*)&bx, sizeof(float)) &&
+          file.read((char*)&by, sizeof(float)) && file.read((char*)&bz, sizeof(float))))
+      break;
+
+    const auto point = Vec3(px, py, pz);
+    if (!isValid(point))
+      continue;
+
+    points.push_back(Vector3f(px, py, pz));
+    referenceB_vec.push_back(GlobalVector(bx, by, bz));
+    count++;
+  } while (count < numberOfPoints);
+
+  const size_t size = points.size();
+  const size_t size_3 = size * 3;
+  // allocate the input and output host buffer in pinned memory accessible by the Platform devices
+  auto in_host = cms::alpakatools::make_host_buffer<Vector3f[], Platform>(size);
+  auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size_3);
+  // fill the input buffers, and the output buffer with zeros
+  for (size_t i = 0; i < size; ++i) {
+    in_host[i] = points[i];
+    for (uint j = 0; j < 3; j++) {
+      out_host[i*3 + j] = 0.;
+    }
+  }
+
+  float resolution = 0.2;
+  float maxdelta = 0.;
+  int fail = 0;
+
+  // run the test on each device
+  for (auto const& device : devices) {
+    auto queue = Queue(device);
+    // allocate input and output buffers on the device
+    auto points_dev = cms::alpakatools::make_device_buffer<Vector3f[]>(queue, size);
+    auto deviceB = cms::alpakatools::make_device_buffer<float[]>(queue, size_3);
+
+    // copy the input data to the device; the size is known from the buffer objects
+    alpaka::memcpy(queue, points_dev, in_host);
+    // fill the output buffer with zeros; the size is known from the buffer objects
+    alpaka::memset(queue, deviceB, 0.);
+
+    auto workDiv = cms::alpakatools::make_workdiv<Acc1D>(1, size);
+    alpaka::exec<Acc1D>(queue, workDiv, MagneticFieldKernel{}, points_dev.data(), deviceB.data(), size);
+
+    // copy the results from the device to the host
+    alpaka::memcpy(queue, out_host, deviceB);
+
+    // wait for the kernel and the potential copy to complete
+    alpaka::wait(queue);
+
+    // check the results
+    for (uint i = 0; i < points.size(); i++) {
+      const auto point = points[i];
```

Comment:
if you don't need to make a copy, better to use a reference:
```suggestion
      const auto& point = points[i];
```

Reply:
Right: https://github.com/cms-sw/cmssw/pull/44360/commits/c6936d662ee52343d6446d101f7660a5c8ddffda